src/commands/editContainerApp.ts

```diff
@@ -0,0 +1,21 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { KnownActiveRevisionsMode } from "@azure/arm-appcontainers";
+import type { IActionContext } from "@microsoft/vscode-azext-utils";
+import { ext } from "../extensionVariables";
+import type { ContainerAppItem } from "../tree/ContainerAppItem";
+import { localize } from "../utils/localize";
+import { pickContainerApp } from "../utils/pickContainerApp";
+
+export async function editContainerApp(context: IActionContext, node?: ContainerAppItem): Promise<void> {
+    node ??= await pickContainerApp(context);
+
+    if (node.containerApp.revisionsMode !== KnownActiveRevisionsMode.Single) {
+        throw new Error(localize('revisionModeError', 'The issued command can only be executed in single revision mode.'));
```

Comment:
Is it possible for a user to encounter this error? Can we setup the context values so that this is unreachable if the container app is in multi revision mode?

Reply:
I'd have to take this command out of the command palette, then I think it would be unreachable 🤔

Reply:
Couldn't you just only show Container Apps in single revision mode in the picker?

Reply:
Hmm, how do you feel about adding a flag on `PickItemOptions` that you pass in when calling `pickContainerApp`?  (Reason being, I think there are still cases in multiple revisions mode where you might still want to be able to pick container app items, so we would probably want to differentiate that condition).

Reply:
We know that most users don't use multiple revision mode. I don't think it's worth having this [huge block of code.](https://github.com/microsoft/vscode-azurecontainerapps/pull/405/files#diff-b845014478b2c4c69e8df5ed166a3c3b9b57a2c6f5ba5adc2b8defc7ec184570R44-R60) For the few people that do try to click on a multiple revision app, I think getting an error immediately isn't the worst experience.

Maybe it  would be nice if our picker actually threw an error "step" so that the wizard would allow you to go back.